pkg/cli/admin/nodeimage/create.go

```diff
@@ -293,15 +296,16 @@ func (o *CreateOptions) Run() error {
 		return err
 	}
 
-	err = o.waitForCompletion(ctx)
-	// Something went wrong during the node-joiner tool execution,
-	// let's show the logs and return an error
-	if err != nil || o.nodeJoinerExitCode != 0 {
-		printErr := o.printLogsInPod(ctx)
-		if printErr != nil {
-			return printErr
+	if err = o.waitForCompletion(ctx); err != nil {
+		// Something went wrong during the node-joiner tool execution
+		o.attachPodLogsToReport(ctx, err)
+	}
+	if o.report.Result.ExitCode != 0 {
+		o.log("command execution failed. Reason: %s", o.report.Result.ErrorMessage)
+		if err = o.saveReport(); err != nil {
+			return err
 		}
-		return fmt.Errorf("image generation error: %v (exit code: %d)", err, o.nodeJoinerExitCode)
+		return kcmdutil.ErrExit
```

Comment:
Shouldn't this return the more detailed o.report.Result.ErrorMessage?

Reply:
It's already logged by `o.log()`.  The `kcmdutil.ErrExit` allows the [CheckErr](https://github.com/openshift/oc/blob/bdce85daf64eda40dd1b604270b85c67a296bc93/vendor/k8s.io/kubectl/pkg/cmd/util/helpers.go#L164) to avoid reprinting it, and moreover to terminate the executable with the default error code